src/mmrelay/config.py

```
@@ -229,217 +150,275 @@ def get_credentials_search_paths(
+    If the platform does not support E2EE or directory creation fails (for example due to permissions), a home-based fallback path is returned and a warning is logged. On other runtime errors raised by the unified store resolver, the error is propagated.
+
+    Returns:
+        The absolute path to the E2EE store directory as a string.
+
+    Raises:
+        RuntimeError: If the unified store resolver raises a RuntimeError that is not handled as a platform-unsupported condition.
+    """
+    try:
+        store_dir = str(get_unified_store_dir())
+        os.makedirs(store_dir, exist_ok=True)
+        return store_dir
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)